plugins/inputs/zfs/zfs_freebsd.go

```diff
@@ -137,15 +183,20 @@ func zpool() ([]string, error) {
 	return run("zpool", []string{"list", "-Hp", "-o", "name,health,size,alloc,free,fragmentation,capacity,dedupratio"}...)
 }
 
+func zdataset(properties []string) ([]string, error) {
+	return run("zfs", []string{"list", "-Hp", "-o", strings.join(properties, ',')}...)
```

Comment:
This has to be
```suggestion
	return run("zfs", []string{"list", "-Hp", "-o", strings.Join(properties, ',')}...)
```
Did you compile and run `make check` on a FreeBSD machine?